daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala

```diff
@@ -200,6 +201,7 @@ private[lf] object PartialTransaction {
     keys = Map.empty,
     globalKeyInputs = Map.empty,
     localContracts = Set.empty,
+    localContractsEver = Set.empty,
```

Comment:
Why do we need to track both `localContracts` and `localContractsEver` at the level of PartialTransaction (I get why we need localContracts in ActiveLedgerState)? Before `localContracts` here corresponded to `localContractsEver` and afaict we never need the other one at this level

Reply:
Indeed I think we just need to rename localContracts to `localContractsEver`.

Reply:
Thanks @nickchapman-da. 
I just think the API in 1.18  is complelty broken, active state should be a field of `PartialTransaction`. 
I think you have made the good choice by not refactoring more, we want the change to be minimal to ease any future backport.

Reply:
> The fields of the ActiveLedgerState are duplicated/flattenned in the PartialTransaction state (with no activeState indirection as we have in main)

This is true for `keys` and `consumedBy`. It is not true for `localContracts`. In 1.18, `localContracts` was not reset on a rollback so it corresponds to `localContractsEver` and there was nothing that tracked `locallyCreatedThisTimeline`.

Now I agree that this is pretty messy and we would be better off storing an `ActiveLedgerState` on the `PartialTransaction` but regardless of which approach we take, we should at least take this as an opportunityt to document these fields clearly.